### PR TITLE
NxForm RSC-374

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-374

This PR is a little code-heavy, so I'm requesting review from outside of just @sonatype/team-plaid .

### CODE REVIEW

File: lib/src/base-styles/_nx-forms.scss
Comment (rpokorny) [inline lib/src/base-styles/_nx-forms.scss:11]: For non-fullscreen submit masks

File: lib/src/components/NxTextInput/NxTextInput.tsx
Comment (rpokorny) [inline lib/src/components/NxTextInput/NxTextInput.tsx:62]: In an `NxForm`, you'll typically want to compute validation errors even for pristine inputs, as they should affect the disabled state of the Submit button (e.g. if there are required fields the button should initially be disabled until they are filled in).  The changes here and the `stateHelpers` make that much more convenient.

File: lib/src/util/validationUtil.ts
Comment (rpokorny) [inline lib/src/util/validationUtil.ts:47]: Realized after posting the PR that this needs docs. I'll add them before this is merged.  TODO

**Review by Blacktiger — APPROVED** [review]
Don't forget your TODO, otherwise looks good to me with only a couple of minor comments. I'm pretty sure this will work well with the way the nxrm codebase currently works.

File: lib/src/util/validationUtil.ts
Comment (rpokorny) [reply lib/src/util/validationUtil.ts:47]: > Realized after posting the PR that this needs docs. I'll add them before this is merged. TODO

d44ef6da

File: lib/src/components/NxForm/NxForm.tsx
Comment (rpokorny) [reply lib/src/components/NxForm/NxForm.tsx]: > I wonder if we might want to customize this message?

3fb58d12

File: lib/src/components/NxTextInput/__tests__/NxTextInput.test.tsx
Comment (rpokorny) [reply lib/src/components/NxTextInput/__tests__/NxTextInput.test.tsx]: > spelling?

e01c4e3c

**Review by glenhunter — APPROVED** [review]
This is really cool, I'll leave it to @Blacktiger to review the more sophisticated parts, but what I understand looks great.

**Pull request merged**
